3.3/.sti/bin/run

```diff
@@ -1,25 +1,112 @@
 #!/bin/bash
 
-function is_gunicorn_installed() {
-    pip show gunicorn
+# This script uses these environment variables:
+#
+# APP_MODULE: Python dotted path to your WSGI application.
+#             If not provided, tries to find the Python path to a 'wsgi.py' file
+#             in the source tree.
+#             That file is present in Django projects by default.
+# APP_CONFIG: Optional configuration file for gunicorn.
+# APP_FILE:   Optional path to Python script to run your application.
+#             Defaults to 'app.py' if it exists.
+#
+# DISABLE_MIGRATE: if not empty, inhibits execution of 'manage.py migrate'.
+
+BIND_ADDR="0.0.0.0:8080"
+
+function run() {
+  # For SCL enablement
+  source .bashrc
+
+  set -e
+
+  # Try to run application with the strategies below, in precedence order.
+  # The first successful strategy takes over this script's process via exec.
+  # If no strategy succeed, report an error message and terminate.
+  run_django
+  run_gunicorn "$APP_MODULE"
+  run_python_script "${APP_FILE:-app.py}"
+
+  echo "ERROR: don't know how to run your application."
+  echo "Please set either APP_MODULE or APP_FILE environment variables, or create a file 'app.py' to launch your application."
+  return 1
 }
 
-# For SCL enablement
-source .bashrc
+function run_django() {
```

Comment:
this should be done in this way:

```sh
if has_migrations; then
  run_django_migrate "${manage_script}"
fi
if has_gunicorn; then
  run_gunicorn "${APP_MODULE}"
else
  run_django_server "${manage_script}"
fi
```

If the run_django_server and run_unicorn are just 2 lines I would nuke that functions.... we should move the **detection** to functions not the actual server run.